### PR TITLE
Fix provider validation when presets are used

### LINKED ISSUE #1583: Renaming a cluster will throw errors
When renaming a cluster an error will appear (at least for digitalocean).

Steps to reproduce:
1. In Cluster Details View click the burger menu on the right in the cluster overview box. A popup will appear
2. Enter a different name and click save.

For Digitalocean you will get following error:
```
{
  "error":{
     "code":400,
     "message":"invalid cluster: invalid cloud spec: GET https://api.digitalocean.com/v2/regions: 401 Unable to authenticate you."
  }
}
```

Just tested on Digitalocean (not working) & AWS (working) so far, please also check all other providers!

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: Fixes https://github.com/kubermatic/dashboard-v2/issues/1583.

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Fixed provider validation when presets are used.
```

### CODE REVIEW

File: api/pkg/provider/cloud/kubevirt/provider.go [hunk 1]
```diff
@@ -2,35 +2,40 @@ package kubevirt
...
+		return err
 	}
 
-	config, err := base64.StdEncoding.DecodeString(spec.Kubevirt.Kubeconfig)
+	config, err := base64.StdEncoding.DecodeString(kubeconfig)
 	if err != nil {
 		// if the decoding failed, the kubeconfig is sent already decoded without the need of decoding it,
 		// for example the value has been read from Vault during the ci tests, which is saved as json format.
-		config = []byte(spec.Kubevirt.Kubeconfig)
+		config = []byte(kubeconfig)
```
Comment (maciaszczykm) [inline api/pkg/provider/cloud/kubevirt/provider.go:38]: @moadqassem Could you take a look if it's okay? 

By the way, I don't fully get the line 46 (previously 41). Why do you update kubeconfig there?

Comment (alvaroaleman) [reply api/pkg/provider/cloud/kubevirt/provider.go:38]: That line pretty much does not do anything, as we get a literal and not a pointer, so looks like a remainder for me.

/approve

**Pull request merged**